kong/plugins/aws-lambda/iam-role-credentials.lua

```diff
@@ -0,0 +1,83 @@
+local http  = require "resty.http"
+local json  = require "cjson"
+local cache = require "kong.tools.database_cache"
+
+local CACHE_IAM_INSTANCE_CREDS_DURATION = 60 -- seconds to cache credentials from metadata service
+local IAM_CREDENTIALS_CACHE_KEY = "plugin.aws-lambda.iam_role_temp_creds"
+
+local function fetch_iam_credentials_from_metadata_service(metadata_service_host, metadata_service_port)
+    local client = http.new()
+    client:set_timeout(500)
```

Comment:
Not quite sure about the validity of this comment, given the very narrow infrastructure scope of the context at play here. Adding a tunable for this seems like unnecessary cognitive load.